app/src/main/java/com/io/movio/data/repositories/MovieRepositoryImpl.kt

```diff
@@ -0,0 +1,55 @@
+package com.io.movio.data.repositories
+
+import com.io.movio.common.Constant.BASE_IMAGE_URL
+import com.io.movio.data.MovieApi
+import com.io.movio.domain.Movie
+import com.io.movio.data.resources.MovieListResource
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
```

Comment:
I don't think you need this Singleton annotation since you have it on top of the provide method

Reply:
true